File: python/coins_and_triangle.py
def rows(n):
    r = 0
    for i in range(1, n+1):
        r += i
        if r == n:
            r = i
            break
        elif r > n:
            r = i - 1
            break

    return r

def main():
    test_cases = int(input())

    for i in range(test_cases):
        n = int(input())
        print(rows(n))

if __name__ == '__main__':
    main()

File: rust/coins_and_triangle.rs
use std::io;

fn int_input() -> isize {
    let mut input = String::new();
    io::stdin()
        .read_line(&mut input)
        .expect("Can't read input");
    
    let input: isize = input.trim().parse().expect("This is not a number");

    input
}

fn determine_rows(coins: &isize) -> isize {
    let mut rows: isize = 0;

    rows
}

fn main() {
    let test_cases: isize = int_input();

    for _i in 0..test_cases {
        let n: isize = int_input(); //number of gold coins
        println!("{}", determine_rows(&n));
    }
}